ethereum/core/src/main/java/org/hyperledger/besu/ethereum/chain/DefaultBlockchain.java

```diff
@@ -520,6 +546,23 @@ public Difficulty calculateTotalDifficulty(final BlockHeader blockHeader) {
     return blockHeader.getDifficulty().add(parentTotalDifficulty);
   }
 
+  @Override
+  public Difficulty calculateTotalDifficultyForSyncing(final BlockHeader blockHeader) {
+    if (blockHeader.getNumber() == BlockHeader.GENESIS_BLOCK_NUMBER) {
+      difficultyForSyncing = blockHeader.getDifficulty();
+    } else if (difficultyForSyncing.equals(Difficulty.ZERO)) {
+      final Difficulty parentTotalDifficulty =
+          blockchainStorage
+              .getTotalDifficulty(blockHeader.getParentHash())
+              .orElseThrow(
+                  () -> new IllegalStateException("Blockchain is missing total difficulty data."));
+      difficultyForSyncing = parentTotalDifficulty.add(blockHeader.getDifficulty());
+    } else {
+      difficultyForSyncing = difficultyForSyncing.add(blockHeader.getDifficulty());
```

Comment:
Is this just an optimised path to avoid db lookup for the parent header?

Reply:
Yes

Reply:
I think the parent header is already cached as we cache blocks when importing so this might be needed